horreum-backend/src/main/java/io/hyperfoil/tools/horreum/changedetection/HunterEDivisiveModel.java

```diff
@@ -0,0 +1,280 @@
+
+                            changePoint.description = String.format("eDivisive change `%s` at `%s` for dataset: %d", change, timestamp, datapointID);
+
+                            log.trace(changePoint.description);
+                            changeConsumer.accept(changePoint);
+                        } else {
+                            log.errorf("Could not find datapoint in set!");
+                        }
+                    } else {
+                        log.errorf("Could not parse hunter line: '%s'", changeDetails);
+                    }
+                }
+            }
+
+        } else {
+            log.debugf("No change points were detected in : %s", tmpFiles.tmpdir.getAbsolutePath());
+        }
+    }
+
+    protected boolean validateInputCsv(TmpFiles tmpFiles) {
+        executeProcess(tmpFiles, true, "bash", "-l", "-c", "hunter validate");
+
+        try(FileReader fileReader = new FileReader(tmpFiles.logFile);
+            BufferedReader reader = new BufferedReader(fileReader);){
+
+            Optional<String> optLine = reader.lines().filter(line -> line.contains("Validation finished")).findFirst();
+            if(optLine.isEmpty()) {
+                log.errorf("Could not validate: %s", tmpFiles.tmpdir.getAbsolutePath());
+                return false;
+            }
+            if( optLine.get().contains("INVALID") ) {
+                log.errorf("Invalid format for: %s; see log for details: %s", tmpFiles.tmpdir.getAbsolutePath(), tmpFiles.logFile.getAbsolutePath());
+                return false;
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        return true;
+    }
+
+    @Override
+    public ModelType getType() {
+        return ModelType.BULK;
+    }
+
+
+    protected static class TmpFiles {
+        final File inputFile;
+        final File tmpdir;
+        final File confFile;
+        final File logFile;
+
+        public static TmpFiles instance() throws IOException {
+            return new TmpFiles();
+        }
+
+        public TmpFiles() throws IOException {
+            tmpdir = Files.createTempDirectory("hunter").toFile();
+
+            Path respourcesPath = Path.of(tmpdir.getAbsolutePath(), "tests", "resources");
+            Files.createDirectories(respourcesPath);
+            inputFile = Path.of(respourcesPath.toFile().getAbsolutePath(), "horreum.csv").toFile();
+
+            confFile = Path.of(respourcesPath.toFile().getAbsolutePath(), "hunter.yaml").toFile();
+            logFile = Path.of(respourcesPath.toFile().getAbsolutePath(), "hunter.log").toFile();
+
+            try (InputStream confInputStream = HunterEDivisiveModel.class.getClassLoader().getResourceAsStream("changeDetection/hunter.yaml")) {
+
+
+                try( OutputStream confOut = new FileOutputStream(confFile)){
+                    confOut.write(confInputStream.readAllBytes());
```

Comment:
If I am not wrong this might produce a `NullPointerException` if `confInputStream` is null, which might happen if for any reason `changeDetection/hunter.yaml` is not found. So maybe it is safer to add a null check on `confInputStream`.

Reply:
I have added a null check